src/main/java/org/aesh/parser/LineParser.java

```diff
@@ -198,7 +198,13 @@ else if (parseCurlyAndSquareBrackets && (c == CURLY_END || c == PARENTHESIS_END)
                     handleCurlyEnd(c);
                 }
                 else if (haveEscape) {
-                    builder.append(BACK_SLASH);
```

Comment:
Seems that the \ shouldn't be removed.

Reply:
As far as I understand the addition of \ is replaced by the lines 201-207. Which add the \ when it is not escaping an operator:
```
if (!isQuoted()
  && (currentOperator = matchesOperators(operators, text, index)) != OperatorType.NONE) {
    // Do not add the \ that was a way to escape an operator.
} else {
    builder.append(BACK_SLASH);
}
```

Otherwise the \  would be inserted 2 times in case it is not escaping an operator.

Reply:
@MMarus , I read too quickly and thought that it was the other parse method. Forget about this comment.